### PR TITLE
Update Advanced Build documentation

### PR DESCRIPTION
**Pull request by blowekamp**
Elaborate on how ITK remote modules can be enabled, and provide an
example of how to do it.

### CODE REVIEW

File: Documentation/docs/source/building.rst [hunk 1]
```diff
@@ -197,14 +197,27 @@ After proper configuration, SimpleITK can be built:
...
-SimpleITK is aware of the enabled and disabled ITK Modules. It is
-possible to customize SimpleITK by disabling unneeded ITK modules or to
-extend SimpleITK by writing custom json file dependent an ITK Remote
-Modules being enabled.
-
-Additionally, each of the Examples directory, and each of the Wrapping
-subdirectories can be configured as independent CMake project dependent
-on an installed SimpleITK.
+SimpleITK is aware of the modularity of ITK and automatically enables
+and disables filters based on which modules are available from the ITK
```
Comment (zivy) [inline Documentation/docs/source/building.rst]: "from the ITK **build** SimpleITK is compiled against."

File: Documentation/docs/source/building.rst [hunk 2]
```diff
@@ -197,14 +197,27 @@ After proper configuration, SimpleITK can be built:
...
+SimpleITK is compiled against. This makes is possible to customize
+SimpleITK to be a small library or to wrap additional ITK remote
+modules simply configuring ITK with the desired modules enabled.
+
+For example, the CoherenceEnhancingDiffusionImageFilter is an optional
+filter in SimpleITK as it's from the ITK remote module
```
Comment (zivy) [inline Documentation/docs/source/building.rst]: ...as it's **part of** the ITK remote module...

File: Documentation/docs/source/building.rst [hunk 3]
```diff
@@ -197,14 +197,27 @@ After proper configuration, SimpleITK can be built:
...
+when building ITK and SimpleITK. To enabled it when using the
+Superbuild add `-DModule_AnisotropicDiffusionLBR:BOOL=ON` to the
+command line or in the CMake GUI press the "Add Entry" button to
+define the variable as above.
+
+SimpleITK has a very flexible and robust build system utilizing
+CMake. It enables packages to build SimpleITK in a variety of ways to
+suit their requirements and minimize recompilation of SimpleITK so
+that it can be wrapped for many different languages. Each of the
+language wrapping sub-directories i.e. "Wrapping/Python" can be
```
Comment (zivy) [inline Documentation/docs/source/building.rst]: e.g. not i.e. (e.g. - for example, i.e. - in other words/that is)

File: Documentation/docs/source/building.rst [hunk 4]
```diff
@@ -197,14 +197,27 @@ After proper configuration, SimpleITK can be built:
...
+configured and build as an independent project which is dependent on
+SimpleITK as an install packaged of is libraries and header filers.
```
Comment (zivy) [inline Documentation/docs/source/building.rst]: ...configured and **built** as an independent project which is dependent on **the installed SimpleITK libraries and header files.**

**Review by zivy — COMMENTED** [review]
Added comments on the text with proposed modifications in bold.

**Comment by blowekamp** [comment]
Thanks for the review.

File: Documentation/docs/source/building.rst [hunk 5]
```diff
@@ -197,14 +197,28 @@ After proper configuration, SimpleITK can be built:
...
-possible to customize SimpleITK by disabling unneeded ITK modules or to
-extend SimpleITK by writing custom json file dependent an ITK Remote
-Modules being enabled.
-
-Additionally, each of the Examples directory, and each of the Wrapping
-subdirectories can be configured as independent CMake project dependent
-on an installed SimpleITK.
+SimpleITK is aware of the modularity of ITK and automatically enables
+and disables filters based on which modules are available from the ITK
+build which SimpleITK is compiled against. This makes is possible to
```
Comment (zivy) [inline Documentation/docs/source/building.rst]: ..makes **it** possible

File: Documentation/docs/source/building.rst [hunk 6]
```diff
@@ -197,14 +197,28 @@ After proper configuration, SimpleITK can be built:
...
+command line or in the CMake GUI press the "Add Entry" button to
+define the variable as above.
+
+SimpleITK has a very flexible and robust build system utilizing
+CMake. It enables packagers to build SimpleITK in a variety of ways to
+suit their requirements and minimize recompilation of SimpleITK so
+that it can be wrapped for many different languages. Each of the
+language wrapping sub-directories e.g. "Wrapping/Python" can be
+configured and built as an independent project which is dependent on
+SimpleITK as an install packaged of is libraries and header filers.
```
Comment (zivy) [inline Documentation/docs/source/building.rst]: ... as an **installed package with its libraries and header files**.

**Pull request merged**